### PR TITLE
Add Libero infrastructure scope ADR

### PR DESCRIPTION
**Pull request by giorgiosironi**
As part of the work of [deploying the first Libero product to Libero infrastructure](https://github.com/libero/publisher-chart) rather than inside eLife, it's helpful to define the reusability of the approach and of the actual code that implements it.

The [`continuous delivery` label](https://github.com/libero/publisher/issues?q=is%3Aissue+is%3Aopen+label%3A%22continuous+delivery%22) collects a series of concerns whose scope will vary significantly if we classify them as out of scope according to this decision, or we need to consider in the reference implementation.

### CODE REVIEW

File: doc/adr/0008-libero-infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## Decision
+
+Libero infrastructure should serve two purposes:
+
+- provide *demo* environments to showcase Libero products in certain configurations
+- provide realistic *reference* environments that do not serve real users but can be forked and adapted by service providers to kick start their Libero offering
+
+## Consequences
+
+Vanilla configuration of Libero products should always be running on Libero infrastructure.
```
Comment (thewilkybarkid) [inline doc/adr/0008-libero-infrastructure.md]:
```suggestion
Demo environments for Libero products are run on Libero infrastructure.
```

File: doc/adr/0008-libero-infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Service providers cater for the disparate, very specific needs of publishers.
+
+Service providers may consolidate their infrastructure with the rest of the platforms.
+
+## Decision
+
+Libero infrastructure should serve two purposes:
+
+- provide *demo* environments to showcase Libero products in certain configurations
+- provide realistic *reference* environments that do not serve real users but can be forked and adapted by service providers to kick start their Libero offering
```
Comment (thewilkybarkid) [inline doc/adr/0008-libero-infrastructure.md:24]: Take 'demo' to be something that _is_ running, and 'reference' to be something that is _not_ running.

Comment (giorgiosironi) [reply doc/adr/0008-libero-infrastructure.md:24]: Would only deploy `reference` to make sure it's working, so could be on each build of whatever Infrastructure As Code repository represents it. At the start it probably coincides with `demo` but can see that they may diverge later.

**Review by thewilkybarkid — APPROVED** [review]

**Review by BlueReZZ — APPROVED** [review]

**Review by NuclearRedeye — APPROVED** [review]

**Review by diversemix — APPROVED** [review]
Looks good, I am left wondering what are the expectations for a demo environment from the point of view of "external relations"... can we truly say that monitoring for instance is out of scope? /cc @BlueReZZ

**Comment by giorgiosironi** [comment]
> the expectations for a demo environment from the point of view of "external relations"... can we truly say that monitoring for instance is out of scope?

We have [black box monitoring](https://synthetics.newrelic.com/accounts/1451451/monitors/4eeab62e-df22-4061-b48a-430ec5fcfcc7) that check if key pages are reachable, but no CPU or requests error rate monitoring which is what we are talking about here.

**Comment by erezmus** [comment]
should we have reference deployments in different clouds (obviously not all, but just to show we can deploy on more than aws)

**Comment by giorgiosironi** [comment]
> should we have reference deployments in different clouds

Seems out of scope but @BlueReZZ to answer.

**Comment by BlueReZZ** [comment]
> should we have reference deployments in different clouds (obviously not all, but just to show we can deploy on more than aws)

Eventually, yes. For this iteration I think it's out of scope.

We don't have much knowledge of other cloud providers and most of the early adopters we have spoken to wanted to use AWS. The ideal case would be that someone wants to use another provider and adapts this then contributes back. I'd rather have a very good AWS example that works well and others can adapt than try our hand at expanding this too soon.

**Review by erezmus — APPROVED** [review]

**Pull request merged**